docs/wiki/EL/evm.md

```diff
@@ -0,0 +1,623 @@
+
+![Memory expansion](../../images/evm/memory-expansion.gif)
+
+Writing a word at a 1-byte offset overflows the initial memory page, triggering an expansion to 2 words (64 bytes or 0x40).
+
+### Reading from memory
+
+`MLOAD` reads a value from memory and pushes it onto the stack.
+
+![MLOAD](../../images/evm/mload.gif)
+
+EVM doesn't have a direct equivalent to `MSTORE8` for reading. You must read the entire word using `MLOAD` and then extract the desired byte using a [mask](<https://en.wikipedia.org/wiki/Mask_(computing)>).
+
+> EVM memory is shown as blocks of 32 bytes to illustrate how memory expansion works. In reality, it is a seamless sequence of bytes, without any inherent divisions or blocks.
+
+## Storage
+
+Storage is designed as a **word-addressed word array**. Unlike memory, storage is associated with an Ethereum account and is **persisted** across transactions as part of the world state.
+
+![EVM Storage](../../images/evm/evm-storage.jpg)
+
+Storage can only be accessed via the code of its associated account. External accounts don't have code and therefore cannot access their own storage.
+
+## Writing to storage
+
+`SSTORE` takes two values from the stack: a storage **slot** and a 32-byte **value**. It then writes the value to storage of the account.
+
+![EVM Storage write](../../images/evm/sstore.gif)
+
+We've been running a contract account's bytecode all this time. Only now we see the account and the world state, and it matches the code inside the EVM.
+
+Again, it’s important to note that storage is not part of the EVM itself, rather the currently executing contract account.
+
+The example above shows only a small section of the account's storage. Like memory, all the values in storage are well-defined as zero.
+
+### Reading from storage
+
+`SLOAD` takes the storage **slot** from the stack and loads its value back onto it.
+
+![EVM Storage read](../../images/evm/sload.gif)
+
+> Notice that the storage value persists between examples, demonstrating its persistence within the world state. Since the world state is replicated across all nodes, storage operations are gas expensive.
+
+## Transaction
+
+A **transaction** is a cryptographically-signed instruction issued by **an external account**, broadcasted to the entire network using [JSON-RPC](/wiki/EL/json-rpc.md).
+
+A transaction contains following fields:
+
+- **nonce ($T_n$)**: An integer value equal to the number of transactions sent by the sender. Nonce is used to:
+
+  - **Prevent replay attack**: Let's say Alice sends 1 ETH to Bob in a transaction, Bob might try to rebroadcast the same transaction into the network to get additional funds from Alice's account. Since the transaction is signed with a unique nonce, EVM will simply reject it if Bob sends it again. Thus safeguarding Alice's account from unauthorized duplicate transactions.
```

Comment:
There is a doc on transaction anatomy worked on #67. It needs a *a LOT* of polish right now but I would suggest moving some of this content there because only certain fields are relevant here

Reply:
I would prefer if we moved the whole transaction section to #67 and then link it here. I had to include it here because some context is required but its actually a tangential topic. Does that sound okay?

Reply:
Sounds perfect!